component-runtime-impl/src/test/java/org/talend/sdk/component/runtime/record/SchemaImplTest.java

```diff
@@ -182,6 +194,115 @@ void testRecordWithMetadataFields() {
+        final Entry arrayEntry = new EntryImpl.BuilderImpl() //
+                .withName("test test") //
+                .withRawName("test test") //
+                .withType(Type.ARRAY) //
+                .withNullable(true) //
+                .withElementSchema(innerArray) //
+                .build();
+
+        final Schema schema = new BuilderImpl() //
+                .withType(Type.RECORD) //
+                .withEntry(metaSpace) //
+                .withEntry(entry1) //
+                .withEntry(entry2) //
+                .withEntry(entry3) //
+                .withEntry(arrayEntry)
+                .build();
+        final List<String> arrayValue = new ArrayList<>();
+        arrayValue.add("value 1");
+        final RecordImpl.BuilderImpl builder = new RecordImpl.BuilderImpl(schema);
+        Record record = builder //
+                .withString(nameSpace, "Aloa space space")
+                .withDateTime(nameSpace + 0, new Date(1234567)) //
+                .withBoolean(nameSpace + 1, Boolean.TRUE) //
+                .withDecimal(nameSpace + 2, BigDecimal.ONE) //
+                .withArray(arrayEntry, arrayValue)
+                .build();
+
+        Assertions.assertEquals("Aloa space space", record.getString(nameSpace));
+        Assertions.assertEquals(Boolean.TRUE, record.getBoolean(nameSpace + 1));
+        Assertions.assertEquals(BigDecimal.ONE, record.getDecimal(nameSpace + 2));
+        Assertions.assertNotNull(record.getArray(String.class, "test test"));
+    }
+
+    @Test
+    void testNameSpace2() {
+        Map<String, String> data = new HashMap<>();
+        data.put("avroCompliant", "value 1");
+        data.put("not avro compliant", "value 2");
+        Schema.Builder schemaBuilder = new BuilderImpl().withType(Type.RECORD);
+        data.keySet()
+                .stream()
+                .forEach(k -> schemaBuilder
+                        .withEntry(new SchemaImpl.EntryImpl.BuilderImpl()
+                                .withName(k)
+                                .withType(Schema.Type.STRING)
+                                .build()));
+
+        Schema schema = schemaBuilder.build();
+        Record.Builder recordBuilder = new RecordImpl.BuilderImpl(schema);
+        data.entrySet().stream().forEach(e -> recordBuilder.withString(e.getKey(), e.getValue()));
+        Record record = recordBuilder.build();
+        Assertions.assertEquals("value 1", record.getString("avroCompliant"));
+        Assertions.assertEquals("value 2", record.getString("not avro compliant"));
+    }
+
+    @Test
+    void testNameSpace3() {
+        final Entry entry1 = new EntryImpl.BuilderImpl() //
+                .withName("1 name a") //
+                .withType(Type.DATETIME) //
+                .withDefaultValue(null) //
+                .build();
+        final Entry entry2 = new EntryImpl.BuilderImpl() //
+                .withName("2 name b") //
```

Comment:
changing with value "2 name b" to "2 name a" is correct but will lead to issues for example...